File: portal-end/backend/schemas/message.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MessageOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    msg_id: int | None = None
    direction: str
    user_id: int | None = None
    reply_to: int
    target_node: int | None = None
    path: list[int] | None = None
    sender: str
    text: str
    status: str
    created_at: datetime

File: portal-end/responder-portal/src/types/message.ts
export const MESSAGE_MAX = 400

export interface PortalMessage {
  id: number
  msg_id: number | null
  direction: 'uplink' | 'downlink' | string
  user_id: number | null
  reply_to: number
  target_node: number | null
  path: number[] | null
  sender: string
  text: string
  status: string
  created_at: string
}

export interface ConversationSummary {
  userId: number
  userName?: string
  lastMessage?: PortalMessage
  messageCount: number
}
